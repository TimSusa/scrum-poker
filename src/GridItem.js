import React, { Fragment } from "react";
import { Button } from "@material-ui/core";
import PropTypes from "prop-types";
import Grid from "@material-ui/core/Grid";

export function GridItem(props) {
  const {
    classes,
    itemArray,
    setVal
  } = props;

  return (
    <Fragment>
      {(itemArray || []).map((item, i) => (
        <Grid item xs={4} key={`${i}-grid-item`}>
          <Button
            variant='contained'
            color='primary'
            className={classes.button}
            onClick={(event) => setVal(event.currentTarget.textContent)}
          >
            {item}
          </Button>
        </Grid>
      ))}
    </Fragment>
  );
}

GridItem.propTypes = {
  classes: PropTypes.object,
  itemArray: PropTypes.array,
  setVal: PropTypes.func
};
