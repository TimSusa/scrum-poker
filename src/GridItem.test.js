import React from 'react';
import renderer from 'react-test-renderer';
import { GridItem } from './GridItem';

describe('GridItem', () => {
  test('snapshot renders', () => {
    const props = {
      classes: {},
      itemArray: [],
      setVal: () => {}
    };

    const component = renderer.create(<GridItem props={props} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
