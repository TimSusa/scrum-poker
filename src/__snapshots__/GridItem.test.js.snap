// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridItem snapshot renders 1`] = `null`;
